src/App.tsx

```diff
@@ -1,17 +1,132 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import { addTodo, deleteTodo, getTodos } from './api/todos';
 import { AuthContext } from './components/Auth/AuthContext';
+import { Todo } from './types/Todo';
+
+import { TodoList } from './components/TodoList/TodoList';
+import { TempTodo } from './components/TempTodo/TempTodo';
+import { Footer } from './components/Footer/Footer';
+import { ErrorNotify } from './components/ErrorNotification/ErrorNotification';
+
+import { ErrorMessage } from './types/ErrorMessage';
+import { ErrorData } from './types/ErrorData';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isError, setIsError] = useState<ErrorData>({
+    status: false,
+    notification: ErrorMessage.None,
+  });
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>([]);
+  const [title, setTitle] = useState('');
+  const [isAdding, setIsAdding] = useState(false);
+  const [deletedTodoIds, setDeletedTodoId] = useState<number[]>([]);
```

Comment:
I'm not sure, but why do you need it? Could you explain , pls?

Reply:
2) title using for input in form (creating new todo):
![image](https://user-images.githubusercontent.com/81638800/199995611-72c0472d-eeb4-4e10-a6e7-9667e0d51a15.png)

Reply:
3) isAdding using for display tempTodo under TodoList, also for disabling input when waiting response from server:
![image](https://user-images.githubusercontent.com/81638800/199996179-cc958031-5f6b-4074-900a-dfabf15bacee.png)
![image](https://user-images.githubusercontent.com/81638800/199996257-b58f0777-ea79-4d2a-b92a-7f9d4bf736bb.png)

Reply:
4. deletedTodoIds using for marking todos which will be removing from TodoList (one todo if we delete todo one by one or all completed todo after click on button Clear completed) for turn on and off loader for such todos.
![image](https://user-images.githubusercontent.com/81638800/199998462-0f889407-c4d7-42c3-9df1-b57e2f167f8e.png)
![image](https://user-images.githubusercontent.com/81638800/199998621-c04bcc6f-5b25-48a0-a756-7987f04807d3.png)
![image](https://user-images.githubusercontent.com/81638800/199998876-e5c5a21f-9407-458f-9030-e2066325febf.png)